dlc-manager/src/manager.rs

```diff
@@ -592,6 +623,82 @@ where
         Ok(())
     }
 
+    /// Manually close a contract with the oracle attestations.
+    pub fn close_confirmed_contract(
+        &mut self,
+        contract_id: &ContractId,
+        attestations: Vec<(usize, OracleAttestation)>,
+    ) -> Result<Contract, Error> {
+        if let Some(Contract::Confirmed(contract)) = self.store.get_contract(contract_id)? {
+            let contract_infos = &contract.accepted_contract.offered_contract.contract_info;
+            let adaptor_infos = &contract.accepted_contract.adaptor_infos;
+
+            // find the contract info that matches the attestations
+            if let Some((contract_info, adaptor_info)) =
+                contract_infos.iter().zip(adaptor_infos).find(|(c, _)| {
+                    c.threshold <= attestations.len()
+                        && c.oracle_announcements
+                            .iter()
+                            .zip(attestations.iter())
+                            .all(|(o, (_, a))| same_nonces(o, a))
+                })
+            {
```

Comment:
I don't think that works if the attestations are from oracles that are not the first ones in a n of m event. Also at the moment you're not using the first element from the `attestations` tuple. I feel it's either we put the burden on the caller and have it pass more information to make the lookup easy (like the `contract_info` index, and maybe do some validation), or we need to do some more complex search (I'd probably go with using a hash table).

Reply:
Good point, will fix. Still working on tests so will make sure to have ones that catch this case

Reply:
Think I fixed it now, hashes all the nonces in the announcement and attestations and finds a contract that has enough matches for the threshold